histFactory/condorTools.py

```diff
@@ -43,9 +43,23 @@ def __init__(self, sampleCfg, execPath, plotConfig, baseDir = ".", rescale = Fal
             elif "db_name" in sample.keys():
                 dbSample = self.getSample(sample["db_name"])
             if dbSample.sampletype == u"SKIM" :
-                files = [ (str(file.lfn), file.nevents) for file in dbSample.files ]
+                files = [ (str(file.lfn), file.nevents, file.event_weight_sum, json.loads(file.extras_event_weight_sum)) for file in dbSample.files ]
             else:
-                files = [ ("/storage/data/cms/" + str(file.lfn), file.nevents) for file in dbSample.files ]
+                files = [ ("/storage/data/cms/" + str(file.lfn), file.nevents, file.event_weight_sum, json.loads(file.extras_event_weight_sum)) for file in dbSample.files ]
+            if "sample_fraction" in sample.keys() and sample["sample_fraction"] < 1:
+                # Truncate file list according to the required fraction
+                # NB : the returned file list does not contain exactly nEvt_tot*fraction event
+                #      it is rounded up to nEvt_tot * fraction + nEvt in next file
+                nEvt_tot = dbSample.nevents
+                nEvt_wanted = int(nEvt_tot * sample["sample_fraction"]) + (nEvt_tot % sample["sample_fraction"] > 0)
```

Comment:
What is the last part of the expression supposed to do? Modulo with non-integers looks strange to me, especially with a boolean operation on it